casbin/enforcer.py

```diff
@@ -85,3 +87,19 @@ def get_permissions_for_user(self, user):
     def has_permission_for_user(self, user, *permission):
         """determines whether a user has a permission."""
         return self.has_policy(*permission)
+
+    def get_implicit_roles_for_user(self, user):
+        roles = self.get_roles_for_user(user)
+        res = copy.copy(roles)
+        for r in roles:
+            res.extend(self.get_implicit_roles_for_user(r))
+        return res
+
+    def get_implicit_permissions_for_user(self, user):
+        roles = self.get_implicit_roles_for_user(user)
```

Comment:
`get_implicit_permissions_for_user` should also have the second parameter `domain`.

Reply:
now added